song_rater/templates/song_rater/song_list.html

```diff
@@ -0,0 +1,38 @@
+<!doctype html>
+<html lang="en">
+<style>
+    th, td {
+        padding: 5px;
+        text-align: left;
+    }
+    .center {
+        margin-left: auto;
+        margin-right: auto;
+    }
+</style>
+<head>
+    <meta charset="utf-8">
+    <title>Song List</title>
+</head>
+<body>
+    <table class="center">
+    <caption>Songs</caption>
+    <tr>
+        <th>Artist</th>
+        <th>Title</th>
+        <th>Average Rating</th>
+        <th>Number of Ratings</th>
+    </tr>
+
+    {% for song in song_table %}
+    <tr>
+        <td>{{ song.artist }}</td>
+        <td>{{ song.title }}</td>
+        <td>{{ song.avg_rating }}</td>
+        <td>{{ song.n_ratings }}</td>
+    </tr>
+    {% endfor %}
```

Comment:
You could make this more dynamic, as in displaying a message when no songs have been added to the site yet. It's not an occurrence that will happen a lot after the site goes into production, but it may be a good opportunity to look into template tags like `for empty endfor`.

If you want, you could even only print the header row if there are songs available by printing them in the first iteration of the template-for loop (there's another tag for that).

Reply:
I added the `empty` template tag to show a message to the user. I also found and now use the `forloop.first ` variable which I assume you referred to.

Reply:
Looks good!